streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,216 @@
+                streamsConfig.getInt(StreamsConfig.RECEIVE_BUFFER_CONFIG),
+                streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG), time);
+
+    }
+
+
+    /**
+     * Cretes a new topic with the given number of partitions and replication factor.
+     *
+     * @param internalTopicConfig
+     * @param numPartitions
+     * @param replicationFactor
+     */
+    public void createTopic(final InternalTopicConfig internalTopicConfig, final int numPartitions, final int replicationFactor, final long windowChangeLogAdditionalRetention) {
+        final Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
+        final Map<String, String> topicConfig = new HashMap<>();
+        for (String key : topicProperties.stringPropertyNames()) {
+            topicConfig.put(key, topicProperties.getProperty(key));
+        }
+        final CreateTopicsRequest.TopicDetails topicDetails = new CreateTopicsRequest.TopicDetails(numPartitions, (short) replicationFactor, topicConfig);
+        final Map<String, CreateTopicsRequest.TopicDetails> topics = new HashMap<>();
+        topics.put(internalTopicConfig.name(), topicDetails);
+
+        final CreateTopicsRequest createTopicsRequest = new CreateTopicsRequest(topics, streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG));
+
+        final ClientResponse clientResponse = sendRequest(createTopicsRequest.toStruct(), ApiKeys.CREATE_TOPICS);
+        final CreateTopicsResponse createTopicsResponse = new CreateTopicsResponse(clientResponse.responseBody());
+        if (createTopicsResponse.errors().get(internalTopicConfig.name()).code() > 0) {
+            throw new StreamsException("Could not create topic: " + internalTopicConfig.name() + ". " + createTopicsResponse.errors().get(internalTopicConfig.name()).name());
+        }
+    }
+
+
+    /**
+     * Send a request to kafka broker of this client. Polls the request for a given number of iterations to receive the response.
+     *
+     * @param request
+     * @param apiKeys
+     */
+    private ClientResponse sendRequest(final Struct request, final ApiKeys apiKeys) {
+
+        final Node brokerNode = kafkaClient.leastLoadedNode(new SystemTime().milliseconds());
+        final String brokerId = Integer.toString(brokerNode.id());
+
+        final SystemTime systemTime = new SystemTime();
+
+        final RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(apiKeys),
+                request);
+
+        final ClientRequest clientRequest = new ClientRequest(systemTime.milliseconds(), true, send, null);
+
+
+        final long readyTimeout = systemTime.milliseconds() + MAX_WAIT_TIME_MS;
+        while (!kafkaClient.ready(brokerNode, systemTime.milliseconds()) && systemTime.milliseconds() < readyTimeout) {
+            kafkaClient.poll(streamsConfig.getLong(StreamsConfig.POLL_MS_CONFIG), systemTime.milliseconds());
```

Comment:
If the broker node is not ready, should we consider picking a different broker instead of tie-ing up one thread doing the while loop here?